jdbc/runtime/src/main/java/io/quarkiverse/configextensions/jdbc/runtime/Repository.java

```diff
@@ -0,0 +1,74 @@
+package io.quarkiverse.configextensions.jdbc.runtime;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.HashMap;
+import java.util.Map;
+
+import io.agroal.api.AgroalDataSource;
+import io.agroal.api.configuration.supplier.AgroalConnectionFactoryConfigurationSupplier;
+import io.agroal.api.configuration.supplier.AgroalConnectionPoolConfigurationSupplier;
+import io.agroal.api.configuration.supplier.AgroalDataSourceConfigurationSupplier;
+import io.agroal.api.security.NamePrincipal;
+import io.agroal.api.security.SimplePassword;
+
+public class Repository {
+    private String url;
+    private String username;
+    private String password;
+
+    public Repository(String url, String username, String password) {
+        this.url = url;
+        this.username = username;
+        this.password = password;
+
+    }
+
+    public synchronized Map<String, String> getAllConfigValues(String table, String keyColumn, String valueColumn)
+            throws SQLException {
+        Map<String, String> result = new HashMap<>();
+
+        try (AgroalDataSource datasource = getDataSource(url, username, password)) {
+
+            String selectAllQuery = new StringBuilder("SELECT conf.").append(keyColumn).append(", conf.")
+                    .append(valueColumn).append(" FROM ").append(table).append(" conf").toString();
+            PreparedStatement selectAll = datasource.getConnection().prepareStatement(selectAllQuery);
+
+            if (selectAll != null) {
+                try (ResultSet rs = selectAll.executeQuery()) {
+                    while (rs.next()) {
+                        result.put(rs.getString(1), rs.getString(2));
+                    }
+                }
+            }
+
+        }
+
+        return result;
+    }
+
+    private AgroalDataSource getDataSource(String url, String username, String password) throws SQLException {
```

Comment:
I'm concerned that we need to create the DS ourselves, but I think there is no way around it, because Config initialises before Quarkus datasources.

Reply:
One option could be to don't provide any configuration from jdbc-config content on initialization, and once the app has booted then use the DS in a standard way. Making configurations on DB only accesible for new configuration requests once the app has started. IMHO it's too much complex for this small extension

Reply:
Yes. It's ok. Let's see how it goes.